### ISSUE TITLE
PDF-A produces lossy result

### ISSUE THREAD

**Issue by YutMarma** [ocrmypdf/OCRmyPDF#1255]
`$ ocrmypdf -O0 input.pdf output.pdf`

To the humble end-user, this should be producing a lossless, though likely-unoptimized, .PDF.

Instead, 
![input](https://github.com/ocrmypdf/OCRmyPDF/assets/17182147/6a48572e-0c9c-4174-834b-057df384672f)
![output](https://github.com/ocrmypdf/OCRmyPDF/assets/17182147/9935377e-7401-4b44-983c-a4cf13dadd71)

Despite requesting --Optimize=0, these are still clearly being run through JPEG compression at some point. Why would I want to convert to JPEG unless I was trying to optimize file size?

Well, no worries, maybe this is user error, let's try...

`$ ocrmypdf -O1 --jpg-quality 100 input.pdf output.pdf`

Nope. Same with --png quality, as well.

In fact, the only way to get an actual lossless product is by manually setting `--pdfa-image-compression lossless`. It seems to me that lossless should be the default, because `-auto` is definitely not working as intended (the source .pdf is all text with very minimal formatting, though it's in greyscale instead of pure 1-bit B&W for legibility - maybe that's what `-auto` looks at when determining whether to unleash the JPEG artifacts...?). 

Basically, my concern here is one of general usability - default arguments should be non-destructive. Granted, the amount of "casual users" - by which I mean "people not clueful enough to figure out how to get things right" - using Python on a Linux box to OCR their .PDFs can probably be counted on one hand that's missing all its fingers, but if even one trove of old documents was inadvertently harmed by not realizing that adding OCR would, non-intuitively, change the appearance of the original document, that's one trove too many :)

**Comment by jbarlow83** [comment]
This behavior and can't be changed because ocrmypdf has always produced PDF/A by default. There are workflows where PDF/A is a legislated requirement.

**Comment by YutMarma** [comment]
Correct me if I'm wrong, but it can in fact be changed, simply by making the default `--pdfa-image-compression` option be `-lossless` unless expressly overridden. Then, these workflows where PDF/A are a legislated requirement are not producing inferior, fuzzy documents.

**Comment by jbarlow83** [comment]
I realized you must be using a version of ocrmypdf/Ghostscript old enough to not have JPEG passthrough. That is why this is not normally an issue for anyone. 

The setting you mentioned converts all JPEG to PNG so it is not that useful to most.

**Comment by jbarlow83** [comment]
Assuming old ghostscript was issue

**Comment by YutMarma** [comment]
@jbarlow83 Just to gently follow up on this - the current LTS for Ubuntu (Jellyfish) does not update Ghostscript beyond 9.55. It's possibly not worth fretting over, since 24.04 is due in six weeks from this post, but I do still feel like principle-of-least-astonishment would dictate expressly assuming `--pdfa-image-compression lossless` unless declared otherwise.
